architectures/firmware/sof-zephyr/rtos_layer/memory_management/heap_sharing.rst

```diff
@@ -3,14 +3,55 @@ Heap sharing
 
 The memory heap can be:
 
--  private - attached to one of the DSPs,
+-  local - used exclusively by a single DSP core,
 -  shared - higher level memory shared across all DSP cores
 
-The shared memories are co-managed by all DSP cores that have access to it.
+.. uml:: images/heaps.pu
+   :caption: Memory Heaps
 
-The data structures needed to manage shared memories are initialized by primary
-core, structure location in memory map is known at the build time and access to
-it is controlled by mutex.
+.. note:: Introduction of MMU will require a separate local application heap per
+   isolated domain.
+
+L1 Cache Coherency
+******************
+
+NOTE: This section applies to systems without L1 cache coherency.
```

Comment:
We probably want to say on Intel architecture only atm, I'm sure others will need this advice too for multicore on xtensa.

Reply:
update the note that it applies to Intel arch